### PR TITLE
APP-1414: Add vector input e2e tests

### PR DESCRIPTION
**Pull request by anjinai**
**2023.03.15_APP-1414: Prime Vector Input E2E Tests**

**Items to Note:**

- This PR is comprised of playwright tests to confirm the following behaviors for Prime Vector Inputs.

Behavior | Scenario
-- | --
Default Vector Input | GIVEN no "type" attribute has been applied to the vector input elementAND no "step" attribute has been applied to the vector input elementAND no "dimensions" attribute has been applied to the vector input elementWHEN the element is renderedTHEN the vector input element should render it's inputs as number (text) typesAND should render three inputs with placeholders of x, y, zAND have a default step of 1
Type: Number | GIVEN a "type" attribute of "number" has been applied to the vector input elementWHEN the element is renderedTHEN the vector input element should render it's inputs as number types
Type: Integer | GIVEN a "type" attribute has been applied to the vector input elementWHEN the element is renderedTHEN the vector input element should render it's inputs with integer valuesAND on arrow up/arrow down within those inputs, the inputs should increment/decrement by integer values (step of 1)
Label: Default | GIVEN a "label" attribute has been applied to the vector input elementWHEN the element is renderedTHEN the vector input element should render that label value to the left of the vector input element
Value | GIVEN a "value" attribute has been applied to the vector input element (string with intended values -> value="1, 2, 3")WHEN the element is renderedTHEN the vector input element should render that value or those values within the inputs
Step | GIVEN a "step" attribute has been applied to the vector input elementWHEN the element is renderedTHEN the rendered inputs should be incremented/decremented by that step value on arrow up/arrow down
On: Input | GIVEN an input's value has been incremented/decrementedTHEN the vector input should emit an "input" event with the new input value
Dimensions: 3 | GIVEN a "dimensions" attribute of "3" has been applied to the vector input elementWHEN the element is renderedTHEN the vector input element should render three inputs with the placeholders "x", "y", and "z"
Dimensions: 4 | GIVEN a "dimensions" attribute of "4" has been applied to the vector input elementWHEN the element is renderedTHEN the vector input element should render four inputs with the placeholders "x", "y", "z", and "w"

<p></p>

### CODE REVIEW

File: playground/vector-input-test.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" href="/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>PRIME Testbed: Select</title>
```
Comment (ehhong) [inline playground/vector-input-test.html:7]: [nit] oops, update to vector input

File: tests/vector-input.spec.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  expect(yTypeValue).toBe('text')
+  const zTypeValue = await vectorInputNumType.locator('input').nth(2).getAttribute("type")
+  expect(zTypeValue).toBe('text')
+
+});
+
+test('Renders three inputs with type number if integer type specified', async ({ page }) => {
+  const vectorInputTypeInteger = page.getByTestId('vector-input-type-integer')
+
+  const xTypeValue  = await vectorInputTypeInteger.locator('input').nth(0).getAttribute("type")
```
Comment (ehhong) [inline tests/vector-input.spec.ts:89]: [opt] you can also test this with the `expect(locator).hasAttribute()` function

File: tests/vector-input.spec.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+  const vectorInputOnInput = page.getByTestId('vector-input-on-input')
+
+  await expect(vectorInputOnInput).toBeVisible()
+
+  const xInput = vectorInputOnInput.locator('input').nth(0)
+  const xInputEvent = waitForCustomEventWithParam(page, 'input', 'value')
+  await xInput.focus()
+  await xInput.press('ArrowUp')
+  expect(xInputEvent).toBeTruthy()
+  expect(await getCustomEventParam(page, 'input', 'value')).toBe('1')
```
Comment (ehhong) [inline tests/vector-input.spec.ts:108]: [q] weird that it only dispatches the one value with no annotation if you adjust one dimension? is there no labeling or default for the other dimensions without input?

File: playground/vector-input-test.html (hunk 1 above)
Comment (anjinai) [reply playground/vector-input-test.html:7]: whoops - thanks!

File: tests/vector-input.spec.ts [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+test('When label is specified, label renders to the left of inputs', async ({ page }) => {
+  const vectorInputLabelLeft = page.getByTestId('vector-input-label')
+
+  await expect(vectorInputLabelLeft).toBeVisible()
+
+  const labelText = await vectorInputLabelLeft.getByRole('paragraph').textContent()
+  expect(labelText).toBe('Label Left')
+
+  // Confirm label is to the left of x input (x input is to the right of label)
+  const xInputPlaceholder = (await page.locator("input:right-of(:text('Label Left'))").nth(1).getAttribute('placeholder'))
```
Comment (ehhong) [inline tests/vector-input.spec.ts:136]: this is a triumph. we will overcome the position-based locators !

**Review by ehhong — APPROVED** [review]
some tiny tiny things but overall lgtm!

File: playground/src/vector-input-test.ts
[the remark this answers is not in the document]
Comment (anjinai) [reply playground/src/vector-input-test.ts:1]: After you merge, I'll pull in your changes and put the appropriate files from my pr into src/pages/

File: tests/vector-input.spec.ts (hunk 4 above)
Comment (anjinai) [reply tests/vector-input.spec.ts:136]: :laughing:

File: tests/vector-input.spec.ts (hunk 3 above)
Comment (anjinai) [reply tests/vector-input.spec.ts:108]: Yeah, that's how it's currently set up - no default for the dimensions without a set input.

**Pull request merged**